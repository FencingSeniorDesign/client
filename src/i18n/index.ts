// Export our i18n instance so it can be imported from './i18n'
export { default } from './i18n';
