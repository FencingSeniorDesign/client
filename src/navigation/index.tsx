// src/navigation/index.tsx
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { createStaticNavigation } from '@react-navigation/native';
import React from 'react';
import { Home } from './screens/Home';
import { EventManagement } from './screens/EventManagement';
import { EventSettings } from './screens/EventSettings';
import { RefereeModule } from './screens/RefereeModule/RefereeModule';
import PoolsPage from './screens/PoolsPage';
import BoutOrderPage from './screens/BoutOrderPage';
import RoundResults from "./screens/RoundResults";
import DEBracketPage from './screens/DEBracketPage';
import DoubleEliminationPage from './screens/DoubleEliminationPage';
import CompassDrawPage from './screens/CompassDrawPage';

const RootStack = createNativeStackNavigator({
  screens: {
    HomeTabs: {
      screen: Home,
      options: {
        title: 'Home',
        headerShown: false,
      },
    },
    EventManagement: {
      screen: EventManagement,
      options: {
        title: 'Edit Tournament',
      },
    },
    EventSettings: {
      screen: EventSettings,
      options: {
        title: 'Event Settings',
      },
    },
    RefereeModule: {
      screen: RefereeModule,
      options: {
        title: 'Referee Module',
      },
    },
    PoolsPage: {
      screen: PoolsPage,
      options: {
        title: 'Pools',
      },
    },
    BoutOrderPage: {
      screen: BoutOrderPage,
      options: {
        title: 'Bout Order',
      },
    },
    RoundResults: {
      screen: RoundResults,
    },
    DEBracketPage: {
      screen: DEBracketPage,
      options: {
        title: 'Single Elimination',
      },
    },
    DoubleEliminationPage: {
      screen: DoubleEliminationPage,
      options: {
        title: 'Double Elimination',
      },
    },
    CompassDrawPage: {
      screen: CompassDrawPage,
      options: {
        title: 'Compass Draw',
      },
    }
  },
});

// Create the static navigation with the stack
export const Navigation = createStaticNavigation(RootStack);