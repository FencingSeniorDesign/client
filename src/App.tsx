// src/App.tsx
import { Assets as NavigationAssets } from '@react-navigation/elements';
import { Asset } from 'expo-asset';
import * as SplashScreen from 'expo-splash-screen';
import * as React from 'react';
import { Navigation, AppNavigator } from './navigation';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { setupTournamentSync } from './data/TournamentDataHooks';
import { initializeDatabase } from './db/DrizzleClient';
import tournamentServer from './networking/TournamentServer';
import { AbilityProvider } from './rbac/AbilityContext';
import { ScoringBoxProvider } from './networking/ble/ScoringBoxContext';
// Import i18n instance
import './i18n';

// Create a client
const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            staleTime: 1000 * 60, // 1 minute
            retry: 4, // Increased default retries for better resilience
        },
    },
});

// Initialize database
initializeDatabase().catch(error => {
    console.error('Error initializing database:', error);
});

// Set up tournament sync with real-time updates
setupTournamentSync(queryClient);

// Share the queryClient with the TournamentServer for server-side cache invalidation
tournamentServer.setQueryClient(queryClient);

Asset.loadAsync([...NavigationAssets, require('./assets/logo.png')]);

SplashScreen.preventAutoHideAsync();

export function App() {
    return (
        <QueryClientProvider client={queryClient}>
            <GestureHandlerRootView style={{ flex: 1 }}>
                {/* Add AbilityProvider for RBAC */}
                <AbilityProvider>
                    {/* Add ScoringBoxProvider for persistent BLE connections */}
                    <ScoringBoxProvider>
                        <AppNavigator />
                    </ScoringBoxProvider>
                </AbilityProvider>
            </GestureHandlerRootView>
        </QueryClientProvider>
    );
}
