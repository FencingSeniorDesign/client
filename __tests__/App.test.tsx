// __tests__/App.test.tsx
import React from 'react';
import { render } from '@testing-library/react-native';
import { App } from '../src/App';

jest.mock('react-native-gesture-handler', () => {
    const { View } = require('react-native');
    return {
        GestureHandlerRootView: View,
        // Stub the default export with a no-op install function.
        default: { install: () => {} },
    };
});

// Add this mock before other imports
jest.mock('react-native-tcp-socket', () => ({
    createServer: jest.fn(),
    connect: jest.fn(),
}));

// Add AsyncStorage mock
jest.mock('expo-sqlite/kv-store', () => ({
    __esModule: true,
    default: {
        getItem: jest.fn(() => Promise.resolve(null)),
        setItem: jest.fn(() => Promise.resolve(null)),
        removeItem: jest.fn(() => Promise.resolve(null)),
        clear: jest.fn(() => Promise.resolve(null)),
    },
}));

// Mock expo modules to avoid executing their real logic during tests
jest.mock('expo-asset', () => ({
    Asset: {
        loadAsync: jest.fn(() => Promise.resolve()),
    },
}));

jest.mock('expo-splash-screen', () => ({
    preventAutoHideAsync: jest.fn(() => Promise.resolve()),
    hideAsync: jest.fn(() => Promise.resolve()),
}));

// Mock the database initialization so it doesn't run real operations
jest.mock('../src/db/DrizzleClient', () => ({
    initializeDatabase: jest.fn(() => Promise.resolve()),
}));

// Mock tournament syncâ€”this just prevents side effects in tests
jest.mock('../src/data/TournamentDataHooks', () => ({
    setupTournamentSync: jest.fn(),
}));

// Mock TournamentServer with __esModule flag so that its default export includes setQueryClient
jest.mock('../src/networking/TournamentServer', () => ({
    __esModule: true,
    default: {
        setQueryClient: jest.fn(),
    },
}));

// Optionally, mock the Navigation component to isolate tests for App.
// Here we render a simple Text element with testID "navigation" in place of AppNavigator.
jest.mock('../src/navigation', () => {
    const React = require('react');
    const { Text } = require('react-native');
    return {
        AppNavigator: (props: any) => <Text testID="navigation">Navigation Component</Text>,
    };
});

describe('App Component', () => {
    it('renders without crashing and displays the Navigation component', () => {
        const { getByTestId, toJSON } = render(<App />);
        // Check that the mocked Navigation component is rendered.
        expect(getByTestId('navigation')).toBeTruthy();
        // Optionally, take a snapshot for regression testing.
        expect(toJSON()).toMatchSnapshot();
    });
});
