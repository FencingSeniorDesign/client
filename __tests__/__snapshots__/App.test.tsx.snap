// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing and displays the Navigation component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    testID="navigation"
  >
    Navigation Component
  </Text>
</View>
`;
